crates/matrix-sdk/src/client/mod.rs

```diff
@@ -2541,6 +2541,13 @@ impl Client {
         let base_room = self.inner.base_client.room_knocked(&response.room_id).await?;
         Ok(Room::new(self.clone(), base_room))
     }
+
+    /// Checks whether the provided `user_id` belongs to an ignored user.
+    pub fn is_user_ignored(&self, user_id: &UserId) -> bool {
+        let raw_user_id = user_id.to_string();
+        let current_ignored_user_list = self.subscribe_to_ignore_user_list_changes().get();
+        current_ignored_user_list.contains(&raw_user_id)
+    }
```

Comment:
`subscribe_to_ignore_user_list_changes` is only “runtime-aware”: it knows the changes that happen during the current run of the SDK. If a user has been ignored in a previous instance of the SDK, you will miss it.

I think it's better to use something like this (not tested):

```rust
let state_store = get_it_from_somewhere();
let list  = state_store.get_account_data_event_static::<IgnoredUserListEvent>()…;
```

The new `is_user_ignored` is nice! That's a good idea. It's just the body of the method that should change.

Also, the method should be on `matrix_sdk_base::Client`, not `matrix_sdk::Client`.